miner/src/pool/client.rs

```diff
@@ -50,6 +50,15 @@ pub trait Client: fmt::Debug + Sync {
 	/// Is transaction with given hash already in the blockchain?
 	fn transaction_already_included(&self, hash: &H256) -> bool;
 
+	/// Perform basic/cheap transaction verification.
+	///
+	/// This should include all cheap checks that can be done before
+	/// actually checking the signature, like chain-replay protection.
+	///
+	/// This method is currently used only for verifying local transactions.
+	fn verify_transaction_basic(&self, t: &transaction::UnverifiedTransaction)
```

Comment:
the signature is a bit different from `verify_transaction` (by reference instead of by value and the doesn't return), which is a bit annoying

Reply:
It is a bit annoying; do you mind elaborating on why it has to be that way?

Reply:
For optimization purposes I guess, there is no need to pass it by value here.

Reply:
Yeah, main verification is actually consuming `UnverifiedTransaction` and producing `SignedTransaction`, hence we pass by value. Basic verification is just a bunch of very cheap checks, and we didn't introduce typesafety around it (maybe we should have :)).